Pixel Game/Assets/Scripts/Enemy scripts/AiAttack.cs

```diff
@@ -38,16 +34,11 @@ IEnumerator StartNewAttacks()
 	{
 		while (true)
 		{
-            if (InRange && readyToAttack && EnemyHp > 0)
+            if (InRange && CurrentWeapon.ReadyToAttack && EnemyHp > 0)
             {
                 if (enablePreAttack)
                     yield return PreAttack();
-                // Play the attack sound
-                if (attackSound != null)
-                    AudioSource.PlayClipAtPoint(attackSound, this.transform.position);
-                readyToAttack = false;  //This will be set to true after `timeBetweenAttack` seconds have passed
-                Attack();
-                yield return WaitForNextAttack();
+                Execute(playerGameObject.transform.position);
```

Comment:
Execute is something i would love to hear more about!

Reply:
Right, so this just executes the attack. I really wanted to call it Attack but the class is already called that.
Here the player's position is passed as argument so that enemies with projectile attacks get correct aim.